debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -90,8 +90,13 @@ protected void validateAndLoadSchemaHistory(CommonConnectorConfig config, LogPos
                     // would like to also verify redo log position exists, but it defaults to 0 which is technically valid
                     throw new DebeziumException("Could not find existing redo log information while attempting schema only recovery snapshot");
                 }
-                LOGGER.info("Connector started for the first time.");
+                LOGGER.info("Connector started with no previous offset for partition '{}'", partition);
                 if (schema.isHistorized()) {
+                    if (((HistorizedDatabaseSchema) schema).getSchemaHistory().storageExists()) {
+                        LOGGER.info("Database schema history storage was found. Connector will use the pre-existing storage. Checking settings for the same.");
+                        ((HistorizedDatabaseSchema) schema).getSchemaHistory().checkStorageSettings();
+                        return;
```

Comment:
possibly instead of `return`, `((HistorizedDatabaseSchema) schema).initializeStorage();` should be in `else` branch.